QueueDown/PipesAndChannels2.cs

```diff
@@ -1,12 +1,16 @@
 ﻿using System.Buffers;
+using System.Diagnostics;
 using System.Diagnostics.Metrics;
 using System.IO.Pipelines;
 using System.Threading.Channels;
 
 partial class Program
 {
-    public static void Pipes3(Pipe pipe, Counter<long> counter, List<Task> tasks, MemoryPool<byte> pool)
+    public static PipeReader Pipes3(Counter<long> counter, List<Task> tasks)
     {
+        // This is the memory pool from Kestrel
+        var pool = new PinnedBlockMemoryPool();
+        var pipe = new Pipe(new(pool));
```

Comment:
This get passed into multiple implementations

Reply:
yeah, this was one of the places I wasn't sure if I was breaking the rules. It depends whether the interface is 'all impls write into a pipe' or all impls 'must produce a PipeReader'. If you mandate all impls use a Pipe then this benchmark isn't so much measuring how fast can you merge streams of data, it measures how fast can you move a single stream of data through a Pipe.